app/service/UserService.scala

```diff
@@ -43,5 +43,12 @@ class UserService @Inject() (
   def getUser(id: String)(implicit hc: HeaderCarrier): Future[Operator] =
     for {
       userDetails <- connector.getUserDetails(id)
-    } yield userDetails
+    } yield {
+      if (userDetails.deleted) {
+        throw new Exception("User with pid " + id + " is deleted")
+      } else {
+        userDetails
+      }
+    }
+
```

Comment:
Following up from my comment on the other PR, I think this should return `Option` and filter out `deleted` users as `None`, i.e.

```scala
connector
  .getUserDetails(id) // The connector method returns Future[Option[Operator]]
  .map(_.filterNot(_.deleted)) // Filter out deleted users in the inner Option
```

Reply:
yeah you're right, done this - thanks a lot!